addons/hr_timesheet/tests/test_timesheet.py

```diff
@@ -438,3 +439,31 @@ def test_create_timesheet_with_archived_employee(self):
 
         with self.assertRaises(UserError):
             timesheet.employee_id = self.empl_employee2
+
+    def test_remaining_hours_on_project(self):
+        ''' Test the remaining hours on the project '''
+
+        self.task1.planned_hours = 4.0
+
+        self.assertEqual(self.project_customer.remaining_hours, 4.0, 'Remaining hours should be 4.0 before adding timesheet')
+
+        self.env['account.analytic.line'].create({
+            'project_id': self.project_customer.id,
+            'task_id': self.task1.id,
+            'name': 'test',
+            'unit_amount': 2,
+        })
+        self.assertEqual(self.project_customer.remaining_hours, 2.0, 'Remaining hours should be 2.0 after adding timesheet')
+
+        task_child = self.env['project.task'].create({
+            'name': 'Task Child',
+            'project_id': self.project_customer.id,
```

Comment:
If it is a subtask normally no need to set the project_id, this field will be computed to have the project of its parent.
```suggestion
```